renderlib/io/FileReaderZarr.cpp

```diff
@@ -123,7 +123,35 @@ FileReaderZarr::jsonRead(const std::string& zarrurl)
       attrs = ::nlohmann::json::object_t();
     }
   }
+  return attrs;
+}
+
+::nlohmann::json
+FileReaderZarr::jsonRead(const std::string& zarrurl)
+{
+  if (m_zattrs.is_object()) {
+    return m_zattrs;
+  }
+
+  bool hasV3 = false;
+
+  // try zarr.json first (indicating Zarr v3)
+  // and then try .zattrs (indicating Zarr v2)
+  nlohmann::json attrs = tryReadJson(zarrurl, "zarr.json");
+  if (attrs.is_object() && !attrs.empty()) {
+    hasV3 = true;
+  } else {
+    attrs = tryReadJson(zarrurl, ".zattrs");
+  }
+  if (attrs.is_object() && !attrs.empty()) {
+    LOG_DEBUG << (hasV3 ? "Zarr v3" : "Zarr v2");
+  } else {
+    LOG_ERROR << "Failed to open either a .zattrs or a zarr.json from the specified location " << zarrurl;
+    return ::nlohmann::json::object_t();
+  }
```

Comment:
Nit: I feel like the control flow on this is reads a little weird? It might be nicer to have the V3 return early instead of rejoining and doing separate logic with a flag.

Reply:
I see your point. that might be a bit more readable, I'll type it in and see how it looks